cmake/platform.cmake

```diff
@@ -60,11 +60,13 @@ omr_add_prefix(OMR_ARCH_DEFINITIONS_PREFIXED ${OMR_C_DEFINITION_PREFIX} ${OMR_AR
 
 
 add_definitions(
-	${OMR_OS_COMPILE_OPTIONS}
 	${OMR_OS_DEFINITIONS_PREFIXED}
 	${OMR_ARCH_DEFINITIONS_PREFIXED}
 )
 
+set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} ${OMR_OS_COMPILE_OPTIONS})
+set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} ${OMR_OS_COMPILE_OPTIONS})
```

Comment:
My understanding is: `CMAKE_<LANG>_FLAGS` is a string of space separated flags. Appending a list this way will insert unwanted semicolons into the compile command. You can check out how we're already appending flags just [below](https://github.com/eclipse/omr/pull/1496/files#diff-9086bbfa8e248778f183939aa77c67a8R72). This PR passed CI, but TBH I don't know why.